### ISSUE TITLE
number_with_delimiter method? (thousands separators)

### ISSUE THREAD

**Issue by ghost** [crystal-lang/crystal#5573]
I searched for some time in the API docs and in the source code but could not find a function to format numbers with thousands separators (something like Rails' [number_to_delimiter](https://apidock.com/rails/ActionView/Helpers/NumberHelper/number_with_delimiter) helper).

I'm using this snippet for now (which I adapted from ActiveSupport's source code):

```crystal
def number_with_separator(number)
  sprintf("%0.2f", number).gsub(/(\d)(?=(\d\d\d)+(?!\d))/, "\\1,")
end

number_with_separator 12345678.09
# => "12,345,678.09"
```
I wonder if something like that shouldn't be included in the standard lib?

**Comment by hugoabonizio** [comment]
I think this belongs on a i18n shard, it's useful for presentations but it differs between localizations.

**Comment by RX14** [comment]
Should `Int#to_s` go in a shard because it uses `.` as a decimal separator? Perhaps providing the very most common format in the stdlib and leaving others to implement i18n if required would be the best.

**Comment by hugoabonizio** [comment]
I'm actually referring to thousands separator `,`, for decimal separator there already is just `sprintf("%0.2f", number)`.

**Comment by straight-shoota** [comment]
Yeah this is related to i18n but the implementation of the formatter itself could very well be in the standard library. It just needs to be configureable as to which characters to use as thousands delimiter and decimal separator.
 A i18n shard could just wrap that method providing these characters based on a locale.

**Comment by hugoabonizio** [comment]
Could use `_` as separator (`12_345_678.09`) to feel more natural, the language itself has this construction.

**Comment by straight-shoota** [comment]
Resolved by #6314

**Comment by HertzDevil** [comment]
Can this be closed now?
